[docs] Fix EAS Update branch promotion flow deployment pattern (#22125)
expo/expo: docs/pages/eas-update/deployment-patterns.mdx

2. **Testing changes**
   - (a) We can test changes with TestFlight and Play Store Internal Track.
   - (b) We can test changes with an internal distribution build.

>    - (c) We can test changes with Expo Go or a [Development Build](https://docs.expo.dev/develop/development-builds/introduction/).

Reviewer: ```suggestion
   - (c) We can test changes with Expo Go or a [Development Build](/develop/development-builds/introduction/).
```

We should stick to the internal URL unless there are some problem with those (i.e. missing`/home/` prefix on S3), otherwise links would be incorrect for different envs and additionally we will fully re-render the Next app on navigation.

>    - (c) We can test changes with Expo Go or a [Development Build](https://docs.expo.dev/develop/development-builds/introduction/).

Reviewer 2: Just a small suggestion: shouldn't use complete URLs otherwise it forces full re-render on navigation.

3. **Publishing updates**
   - (a) We can publish updates to a single branch.
   - (b) We can create update branches that are environment-based, like "production" and "staging".
   - (c) We can create update branches that are version-based, like "version-1.0", which enables us to promote updates from one channel to another.

> We can mix and match and tweak the parts above to create a process that is the right balance of release cadence and safety for our team and users.

Reviewer 3: ```suggestion
We can mix, match and tweak the parts above to create a process that is the right balance of release cadence and safety for our team and users.
```

Another trade-off to consider is the amount of bookkeeping of versions/names/environments we'll have to do throughout the process. The less bookkeeping we have to do will make it easier to follow a consistent process. It'll also make it easier to communicate with our colleagues. If we need fine-grained control, bookkeeping will be required to get the exact process we want.

Below, we've outlined four common patterns on how to deploy a project using EAS Update.

## Two-command flow

This flow is the simplest and fastest flow, with the fewest amount of safety checks. It's great for trying out Expo and for smaller projects. Here are the parts of the deployment process above that make up this flow:

**Creating builds:** (a) Create builds for production use only.

> **Testing changes:** (c) Test changes with Expo Go or a [Development Build](https://docs.expo.dev/develop/development-builds/introduction/).

Reviewer: ```suggestion
**Testing changes:** (c) Test changes with Expo Go or a [Development Build](/develop/development-builds/introduction/).
```

Same reason as above.

### Explanation of flow

> 1. Develop a project locally and test changes in Dev Client or Expo Go.

Reviewer 4: ```suggestion
1. Develop a project locally and test changes in your development build or Expo Go.
```

#### Disadvantages of this flow

> - There are no pre-production checks to make sure the code will function as intended. We can test with Expo Go or a [Development Build](https://docs.expo.dev/develop/development-builds/introduction/), but this is less safe than having a dedicated test environment.

Reviewer: ```suggestion
- There are no pre-production checks to make sure the code will function as intended. We can test with Expo Go or a [Development Build](/develop/development-builds/introduction/), but this is less safe than having a dedicated test environment.
```

### Explanation of flow

> 1. Develop a project locally and test changes in Expo Go or a [Development Build](https://docs.expo.dev/develop/development-builds/introduction/).

Reviewer: ```suggestion
1. Develop a project locally and test changes in Expo Go or a [Development Build](/develop/development-builds/introduction/).
```

> 2. Create builds with channels named "production-rtv-1" (indicating a channel with a runtime version 1), which will eventually get reviewed and become available on app stores. Create another set of builds with channels named "staging", which will be used for testing on TestFlight and the Play Store Internal Track.

Reviewer 4: ```suggestion
2. Create builds with channels named "production-rtv-1" (indicating a channel with runtime version "1"), which will eventually get reviewed and become available on app stores. Create another set of builds with channels named "staging", which will be used for testing on TestFlight and the Play Store Internal Track.
```

3. Set up `expo-github-action` to publish updates when merging commits to branches.
4. Merge changes into a branch named "version-1.0".
5. Use the website or EAS CLI to point the "staging" channel at the EAS Update branch "version-1.0". Test the update by opening the apps on TestFlight and the Play Store Internal Track.
6. When ready, use the website or EAS CLI to point the "production-rtv-1" channel at the EAS Update branch "version-1.0".
7. Then, there are two update scenarios you may encounter:
    - A new release does not require a new runtime version:

>         1. Create another GitHub branch named "version-1.1".

Reviewer 3: ```suggestion
        0. Your runtime version of `1` remains unchanged. 
        1. Create another GitHub branch named "version-1.1".
```
Just to make it clear, since we don't conform to semver. Otherwise, I am afraid users may think it's possible to change their rtv from `1.0` to `1.1` and expect the same update to be delivered

Reviewer 4: How about just using version strings like "1" and "2" instead of "1.1"?

        2. Use the website or EAS CLI to point the "staging" channel at the EAS Update branch "version-1.1".
        3. Merge commits into the "version-1.1" branch until the new features and fixes are ready and stable.
        4. Use the website or EAS CLI to point the "production-rtv-1" channel at the EAS Update branch "version-1.1". This will mean that everyone with a production build (users who downloaded the app from the app stores) will now get the latest update on the "version-1.1" branch.
    - A new release requires a new runtime version (for example, when new native libraries are added or SDK version is upgraded):

>         1. Create a new "staging" build with the new runtime version.

Reviewer 3: ```suggestion
        0. You bump your runtime version from `1` to `2`.
        1. Create a new "staging" build with the new runtime version.
```
Just to be explicit about rtv

>         1. Create another GitHub branch named "version-1.1".

Reviewer 3: ```suggestion
        2. Create another GitHub branch named "version-2.0".
```
Implies that it's incompatible with version-1.0 in your previous example

>         2. Use the website or EAS CLI to point the "staging" channel at the EAS Update branch "version-1.1".

Reviewer 3: ```suggestion
        2. Use the website or EAS CLI to point the "staging" channel at the EAS Update branch "version-2.0".
```

>         3. Merge commits into the "version-1.1" branch until the new features and fixes are ready and stable.

Reviewer 3: ```suggestion
        3. Merge commits into the "version-2.0" branch until the new features and fixes are ready and stable.
```

        4. Create a new build with channel named "production-rtv-2", which will eventually get reviewed and become available on app stores.

>         5. Use the website or EAS CLI to point the "production-rtv-2" channel at the EAS Update branch "version-1.2". This will mean that everyone who previously had a production build (users who downloaded the app from the app stores) will continue to get the latest update on EAS Update branch "version-1.1" until they download the new version of the app from the app store, at which time they will get the latest update on EAS Update branch "version-1.2".

Reviewer 3: ```suggestion
        5. Use the website or EAS CLI to point the "production-rtv-2" channel at the EAS Update branch "version-2.0". This will mean that everyone who previously had a production build (users who downloaded the app from the app stores) will continue to get the latest update on EAS Update branch "version-1.1" until they download the new version of the app from the app store, at which time they will get the latest update on EAS Update branch "version-2.0".
```

#### Advantages of this flow

> - This flow is safer than the other flows. All updates are tested on test builds which are distributed to internal tests, and branches are moved between channels, so the exact artifact tested is the one deployed to production builds.

Author: ```suggestion
- This flow is safer than the other flows. All updates are tested on test builds which are distributed to internal testers, and branches are moved between channels, so the exact artifact tested is the one deployed to production builds.
```

#### Disadvantages of this flow:

- You'll have to run two commands instead of one to fix changes on both platforms.

## Branch promotion flow

> This flow is an example of a flow for managing versioned releases. It requires a bit more bookkeeping and limits the ability to use auto-runtime version policies (sdkVersion, appVersion). Here are the parts of the deployment process above that make up this flow:

Reviewer 4: ```suggestion
This flow is an example of a flow for managing versioned releases. 

> **warning** It requires a bit more bookkeeping and does not support automatic [runtime version policies](https://docs.expo.dev/eas-update/runtime-versions/#setting-runtimeversion) (`"sdkVersion"`, `"appVersion"`, and `"nativeVersion"`). You will need to [manually specify](https://docs.expo.dev/eas-update/runtime-versions/#custom-runtimeversion) your runtimes' versions with this flow.

Here are the parts of the deployment process above that make up this flow:
```
